components/script/dom/mouseevent.rs

```diff
@@ -237,10 +262,55 @@ impl MouseEvent {
     pub(crate) fn point_in_target(&self) -> Option<Point2D<f32>> {
         self.point_in_target.get()
     }
+
+    /// Create a [MouseEvent] triggered by the embedder
+    pub(crate) fn for_platform_mouse_event(
+        event: embedder_traits::MouseButtonEvent,
+        pressed_mouse_buttons: u16,
+        window: &Window,
+        hit_test_result: &CompositorHitTestResult,
+        can_gc: CanGc,
+    ) -> DomRoot<Self> {
+        let mouse_event_type_string = match event.action {
+            embedder_traits::MouseButtonAction::Click => "click",
+            embedder_traits::MouseButtonAction::Up => "mouseup",
+            embedder_traits::MouseButtonAction::Down => "mousedown",
+        };
+
+        let client_x = hit_test_result.point_in_viewport.x as i32;
+        let client_y = hit_test_result.point_in_viewport.y as i32;
+        let click_count = 1;
+        let mouse_event = MouseEvent::new(
```

Comment:
I remember there's feedback on the new method of UIEvent/MouseEvent/Pointer has too many arguments.
Do you think it's good to update some patterns in this PR?

Reply:
I tried, but it's a bit of a pain since our event initialization doesn't follow the spec. I do still intend to fix that, but I'd rather do that in a seperate PR.